qcodes_contrib_drivers/drivers/Lakeshore/Model_625.py

```diff
@@ -0,0 +1,523 @@
+        -------
+            status
+        """
+        status, current_step_limit = self._get_quench_detection_setup()
+        return status
+
+
+    def _get_quench_current_step_limit(self) -> float:
+        """
+        Gets current step limit for quench detection
+
+        Returns
+        -------
+            <rate>
+        """
+        status, current_step_limit = self._get_quench_detection_setup()
+        return current_step_limit
+
+
+    def _get_coil_constant(self) -> float:
+        """
+        Gets magnetic field constant in either T/A or kG/A depending on units
+
+        Returns
+        -------
+            <constant>
+        """
+        coil_constant_unit, coil_constant = self._get_field_setup()
+        return coil_constant
+
+
+    def _get_coil_constant_unit(self) -> str:
+        """
+        Gets the units of the magnetic field constant: 0 = T/A, 1 = kG/A
+
+        Returns
+        -------
+            <units>
+        """
+        coil_constant_unit, coil_constant = self._get_field_setup()
+        return coil_constant_unit
+
+
+    def _get_field_ramp_rate(self) -> float:
+        """
+        Gets the field ramp rate in units of T/min
+
+        Returns
+        -------
+            field_ramp_rate (T/min)
+        """
+        coil_constant_unit, coil_constant = self._get_field_setup() # in T/A by default
+        current_ramp_rate = self.current_ramp_rate()    # in A/s
+        field_ramp_rate = current_ramp_rate * coil_constant * 60 # in T/min
+        return field_ramp_rate
+
+
+    def _get_ramping_state(self) -> str:
+        """
+        Gets the ramping state of the power supply (corresponds to blue LED on panel)
+        Is inferred from the status bit register
+
+        Returns
+        -------
+            ramping state
+        """
+        operation_condition_register = self.ask('OPST?')
+        bin_OPST = bin(int(operation_condition_register))[2:]
+        if len(bin_OPST)<2:
+            rampbit = 1
+        else:
+            # read second bit, 0 = ramping, 1 = not ramping
+            rampbit = int(bin_OPST[-2])
+        if rampbit == 1:
+            return 'not ramping'
+        else:
+            return 'ramping'
+
+
+    def _get_operational_errors(self) -> str:
+        """
+        Error Status Query
+
+        Returns
+        -------
+            error status
+        """
+        error_status_register = self.ask('ERST?')
+        # three bytes are read at the same time, the middle one is the operational error status
+        operational_error_registor = error_status_register.split(',')[1]
+        
+        #prepend zeros to bit-string such that it always has length 9
+        oer_bit_str = bin(int(operational_error_registor))[2:].zfill(9)
+        return oer_bit_str
+
+
+    def _get_oer_quench_bit(self) -> int:
+        """
+        Returns the oer quench bit
+
+        Returns
+        -------
+            quench bit
+        """
+        return self._get_operational_errors()[3]
```

Comment:
Mypy complains since this returns a str but you expect an int. Either convert the output to int before returning or change the signature.

Reply:
Thanks, I saw the error. Took me some time to find the output of mypy. I converted to int now.